src/core/execute.c

```diff
@@ -708,22 +716,47 @@ static int restore_confirm_stdio(int *saved_stdin, int *saved_stdout) {
         return r;
 }
 
-static int ask_for_confirmation(const char *vc, char *response, char **argv) {
+static int ask_for_confirmation(const char *vc, const char *cmdline) {
         int saved_stdout = -1, saved_stdin = -1, r;
-        _cleanup_free_ char *line = NULL;
-
+        char c;
+
+        /* Returns:
+         *  -1 - don't execute and pretend that the command failed
+         *   0 - don't execute and pretend that the command succeeed
+         *   1 - positive answer, execute the command
```

Comment:
Hmm, I really don't like APIs that return "magic" integers like this. Either:

* Define a proper enum for this, and return these enum values (and if you like, you can even define them to the exact same values -1, 0, 1 to make this pretty)
* or, make this another function that returns boolean-or-error. i.e. return 0 for false, > 0 for true, and < 0 as proper negative errno. If you pick this you'd just propagate errors properly, and potentially synthesize your own, for example -ECANCELED or so.